packages/lesswrong/components/posts/TableOfContents/MultiToCLayout.tsx

```diff
@@ -129,22 +129,24 @@ const MultiToCLayout = ({segments, classes, tocRowMap = [], showSplashPageHeader
   tocRowMap?: number[], // This allows you to specify which row each ToC should be in, where maybe you want a ToC to span more than one row
   showSplashPageHeader?: boolean,
 }) => {
+  const [leftHover, setLeftHover] = React.useState(false);
   const tocVisible = true;
   const gridTemplateAreas = segments
     .map((_segment,i) => `"... toc${tocRowMap[i] || i} gap1 content${i} gap2 rhs${i} gap3 ..."`)
     .join('\n')
   return <div className={classNames(classes.root)} style={{ gridTemplateAreas }}>
     {segments.map((segment,i) => <React.Fragment key={i}>
       {segment.toc && tocVisible && <>
-        <div className={classNames(classes.toc, { [classes.commentToCMargin]: segment.isCommentToC, [classes.splashPageHeaderToc]: showSplashPageHeader, [classes.normalHeaderToc]: !showSplashPageHeader })} style={{ "gridArea": `toc${i}` }} >
+        <div className={classNames(classes.toc, { [classes.commentToCMargin]: segment.isCommentToC, [classes.splashPageHeaderToc]: showSplashPageHeader, [classes.normalHeaderToc]: !showSplashPageHeader })} style={{ "gridArea": `toc${i}` }} onMouseEnter={() => setLeftHover(true)} onMouseLeave={() => setLeftHover(false)}>
           <div className={classNames(classes.stickyBlockScroller, { [classes.commentToCIntersection]: segment.isCommentToC })}>
             <div className={classes.stickyBlock}>
-              {segment.toc}
+              {/* This allows the ToC to appear when the user hovers over either of the two left-columns */}
+              {React.cloneElement(segment.toc as React.ReactElement, { hover: leftHover })}
```

Comment:
idk what this is doing but i'm scared of it

Reply:
I updated the comments to be more clear. (I got this from ChatGPT, but, it makes conceptual sense to me. We need to pass a parameter into a component we already declared in a parent-component, and this seems like a reasonable way to do it)

Reply:
I really feel like this can't be the right way of doing this. Does it just work if you create a new capitalized variable that's assigned to segment.toc and then use the standards JSX syntax for this?

Reply:
I'm not sure what you mean but I just tried my guess (const TOC = toc; <TOC hover={hover/>), which didn't work. Toc isn't a component function, it's already a rendered component.

I asked ChatGPT for variations on this solution and all the alternatives were more lines or clunkier.